ttnn/cpp/ttnn/operations/normalization/layernorm_distributed/layernorm_post_all_gather.cpp

```diff
@@ -15,17 +16,31 @@ ttnn::Tensor ExecuteLayerNormPostAllGather::invoke(
     const std::optional<const ttnn::Tensor>& weight,
     const std::optional<const ttnn::Tensor>& bias,
     const std::optional<MemoryConfig>& memory_config,
-    const std::optional<const DeviceComputeKernelConfig> compute_kernel_config) {
+    const std::optional<const DeviceComputeKernelConfig> compute_kernel_config,
+    const std::optional<const LayerNormProgramConfig>& program_config) {
     auto arch = input_tensor.storage_type() == StorageType::DEVICE ? input_tensor.device()->arch() : ttnn::operations::experimental::auto_format::AutoFormat::GetDefaultDevice()->arch();
     auto kernel_config_val = init_device_compute_kernel_config(arch, compute_kernel_config, MathFidelity::HiFi4, true, false, false);
-    return operation::run(
-                LayerNormPostAllGather{
-                    .norm_type = LayerNormDistributedType::LAYERNORM,
+    if (input_tensor.is_sharded()) {
```

Comment:
@johanna-rock-tt , here is the conditions to choose interleaved or sharded op based on input mem config